src/ado/api/index.ts

```diff
@@ -1,6 +1,47 @@
 import { BatchWorkItems, Iteration, IterationWorkItems, ListIteration, WorkItem, WorkItemHistory } from "../../models/adoApi";
 import { AdoConfigData } from "../../models/adoConfig";
 
+async function fetchWithAuth(url: string, retry: number = 0): Promise<any> {
+  try {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    if (error instanceof SyntaxError && retry < 3) {
+      console.error(`JSON Parse error. Attempting to authenticate. Retry ${retry + 1} / 3`);
+      // JSON parse error indicates unauthenticated request
+      await authenticate(url);
```

Comment:
Is it possible it will get stuck in an await and not return, therefore causing a hang?

Reply:
At the end of the day, this is waiting on some `fetch` calls. I researched this online to see what the default timeout is and it turns out it depends on the browser. For Chrome, it's 300 seconds.

So while this should be transparent to the user you are right to call this out. We should enforce a smaller timeout window for fetch. I'll do that on another p/r as it's a little bit more complex, and I want to make sure I address all API calls we're performing.